hypha/apply/funds/models/submissions.py

```diff
@@ -567,6 +574,11 @@ def ensure_user_has_account(self):
                     email=email,
                     defaults={'full_name': full_name}
                 )
+                # Ensure applying user should have applicant role
+                applicant_group = Group.objects.get(name=APPLICANT_GROUP_NAME)
+                if applicant_group not in self.user.groups.all():
```

Comment:
```suggestion
                if not self.user.is_applicant:
```

Use `is_applicant` method